lib/datadog_backup/core.rb

```diff
@@ -73,12 +73,38 @@ def myclass
       self.class.to_s.split(':').last.downcase
     end
 
+    # Calls out to Datadog and checks for a '200' response
+    def create(body)
+      result = with_200 do
+        api_service.request(Net::HTTP::Post, "/api/#{api_version}/#{api_resource_name}", nil, body, true)
+      end
+      logger.warn 'Successfully created in datadog.'
+      result
+    end
+
     # Calls out to Datadog and checks for a '200' response
     def update(id, body)
-      with_200 do
+      result = with_200 do
         api_service.request(Net::HTTP::Put, "/api/#{api_version}/#{api_resource_name}/#{id}", nil, body, true)
       end
       logger.warn 'Successfully restored to datadog.'
+      result
+    end
+
+    def restore(id)
+      body = load_from_file_by_id(id)
+      begin
+        update(id, body)
+      rescue RuntimeError => e
+        if e.message.include?('Request failed with error ["404"')
+          new_id = create(body).fetch('id')
+
+          get_and_write_file(new_id)
+          FileUtils.rm(find_file_by_id(id))
```

Comment:
If new_id == id, then we would end up with no backup post restore. We should perform the rm operation first.